### PR TITLE
Ad400x fmcz

### PR DESCRIPTION
**Pull request by cristiantpop**
Move project in projects folder
Update project with new SPI engine
Add Makefile

### CODE REVIEW

**Comment by mchindri** [comment]
Shouldn't ad400x-fmcz folder be removed? There are the spi_engine files inthere.

File: projects/ad400x-fmcz/src/ad400x_fmcz.c [hunk 1]
```diff
@@ -8,60 +8,57 @@
...
+#define AD400x_EVB_SAMPLE_NO			1000
 #define AD400X_DMA_BASEADDR             XPAR_AXI_AD40XX_DMA_BASEADDR
-#define AD400X_SPI_ENGINE_BASEADDR      XPAR_SPI_AD40XX_AXI_BASEADDR
+#define AD400X_SPI_ENGINE_BASEADDR      XPAR_SPI_ADC_AXI_REGMAP_BASEADDR
 #define AD400x_SPI_CS                   0
 
 #define SPI_ENGINE_OFFLOAD_EXAMPLE	1
 
+struct spi_engine_init_param spi_eng_init_param  = {
+	.type = SPI_ENGINE,
```
Comment (mchindri) [inline projects/ad400x-fmcz/src/ad400x_fmcz.c:25]: This comment is not related to this PR.  
If we select the spi_engine using spi_eng_platform_ops, should we have this type parameter?

File: projects/ad400x-fmcz/src/ad400x_fmcz.c [hunk 2]
```diff
@@ -8,60 +8,57 @@
...
+	.cs_delay = 2,
+	.data_width = 32,
+};
+
 struct ad400x_init_param ad400x_init_param = {
-	/* SPI engine*/
-	{
-		AD400X_SPI_ENGINE_BASEADDR,	/* adc_baseaddr */
-		AD400x_SPI_CS,			/* chip_select */
-		2,				/* cs_delay */
```
Comment (buha) [inline projects/ad400x-fmcz/src/ad400x_fmcz.c]: where do all these parameters go ? does the new engine not support them ?

**Comment by cristiantpop** [comment]
Changelog v1 -> v2:

- Remove old SPI engine

File: projects/ad400x-fmcz/src/ad400x_fmcz.c [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+		mdelay(2000);
+
+		offload_data = (uint32_t *)msg.rx_addr;
+
+		for(i = 0; i < AD400x_EVB_SAMPLE_NO; i++) {
+			data = *offload_data & 0xFFFFF;
+			if (data > 524287)
+				data = data - 1048576;
+			printf("ADC%d: %d\n", i, data);
```
Comment (amiclaus) [inline projects/ad400x-fmcz/src/ad400x_fmcz.c:104]: Codacy complains about printf format for `i`. Maybe you can use `%u` or `%ul`.

File: projects/ad400x-fmcz/src/ad400x_fmcz.c [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+#include "xil_printf.h"
+#include "spi_engine.h"
+#include "ad400x.h"
+#include "error.h"
+#include "delay.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define AD400x_EVB_SAMPLE_NO			1000
```
Comment (amiclaus) [inline projects/ad400x-fmcz/src/ad400x_fmcz.c:17]: Maybe add these macro definitions in a separate file `parameters.h` as it is in the rest of the projects?

File: projects/ad400x-fmcz/src/ad400x_fmcz.c [hunk 5]
```diff
@@ -78,30 +79,25 @@ int main()
...
+		msg.tx_addr = 0xA000000;
+		msg.commands_data = commands_data;
 
-		spi_eng_offload_load_msg(dev->spi_desc, msg);
-		spi_eng_transfer_multiple_msgs(dev->spi_desc, 1000);
+		ret = spi_engine_offload_transfer(dev->spi_desc, msg, AD400x_EVB_SAMPLE_NO);
+		if (ret != SUCCESS)
+			return ret;
 
 		mdelay(2000);
```
Comment (adrimbarean) [inline projects/ad400x-fmcz/src/ad400x_fmcz.c:96]: Is there a better way to detect when the transaction is over? Like an interrupt from the DMA?

File: drivers/adc/ad400x/ad400x.c [hunk 6]
```diff
@@ -61,23 +61,13 @@ int32_t ad400x_spi_reg_read(struct ad400x_dev *dev,
 {
 	int32_t ret;
 	uint8_t buf[2];
-	uint32_t spi_clk_hz_tmp = (uint32_t) -1;
 
 	buf[0] = AD400X_READ_COMMAND;
 	buf[1] = 0xFF;
 
-	// register access runs at a lower clock rate (~2MHz)
```
Comment (mirceacaprioru) [inline drivers/adc/ad400x/ad400x.c:69]: if these are removed how will the driver be able to read registers ?
